src/System.Threading.Tasks/tests/Task/TaskContinueWithTests.cs

```diff
@@ -1211,34 +1211,49 @@ public static void RunLazyCancellationTests_Negative()
         }
 
         [Fact]
-        public static void RunStackGuardTests()
+        public static void LongContinuationChain_ContinueWith_DoesNotStackOverflow()
         {
-            const int DIVE_DEPTH = 12000;
+            const int DiveDepth = 12_000;
 
-            // Test stack guard with ContinueWith.
+            var tcs = new TaskCompletionSource<bool>();
+            var t = (Task)tcs.Task;
+            for (int i = 0; i < DiveDepth; i++)
             {
-                Func<Task, Task> func = completed => completed.ContinueWith(delegate { }, TaskContinuationOptions.ExecuteSynchronously);
-                var tcs = new TaskCompletionSource<bool>();
-                var t = (Task)tcs.Task;
-                for (int i = 0; i < DIVE_DEPTH; i++) t = func(t);
-                tcs.TrySetResult(true);
-                t.Wait();
+                t = t.ContinueWith(_ => { }, CancellationToken.None, TaskContinuationOptions.ExecuteSynchronously, TaskScheduler.Default);
             }
+            tcs.TrySetResult(true);
+            t.Wait();
+        }
+
+        [Fact]
+        public static void LongContinuationChain_Unwrap_DoesNotStackOverflow()
+        {
+            const int DiveDepth = 12_000;
 
-            // Test stack guard with Unwrap
+            Func<long, Task<long>> func = null;
```

Comment:
>null; [](start = 42, length = 5)

do we need this null initialization? can't we assign func directly to the definition next line?

Reply:
It's necessary. It's captured into a lambda, and the compiler needs to know that it's definitely assigned.